### PR TITLE
update renovate config to add labels and add renovate.json validator

### PR DESCRIPTION
**Pull request by jacobdonenfeld**
- Any security PR's from renovate will get tagged as "security".
- On any future edits to the renovate.json, will run a job to validate the configuration to allow the PR to merge.

### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [x] The title of this PR ends with a link to the main issue being address in paranthesis, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [x] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keyworkds](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [x] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [x] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [x] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

**Comment by thedadams** [comment]
@jacobdonenfeld Can you add a note describing why these changes are necessary?

File: .github/workflows/validate-renovate.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: 'Validate renovate config'
+
+on:
+  push:
+    branches:
+      - 'master'
```
Comment (tylerslaton) [inline .github/workflows/validate-renovate.yml:6]: This should be `main`, right?
```suggestion
      - 'main'
```

File: .github/workflows/validate-renovate.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    paths:
+      - 'env/renovate.json'
```
Comment (tylerslaton) [inline .github/workflows/validate-renovate.yml:8]: Should this be `.github/rennovate.json`? I'm not understanding why it changes here.
```suggestion
      - '.github/renovate.json'
```

**Review by tylerslaton — CHANGES_REQUESTED** [review]
Thanks for the updated description! Couple of comments.

**Review by tylerslaton — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Pull request merged**